agent/consul/state/catalog.go

```diff
@@ -2907,6 +2907,36 @@ func serviceGatewayNodes(tx *txn, ws memdb.WatchSet, service string, kind struct
 	return maxIdx, ret, nil
 }
 
+// metricsProtocolForIngressGateway determines the protocol that should be used when fetching metrics for an ingress gateway
+// Since ingress gateways may have listeners with different protocols, favor capturing all traffic by only returning HTTP
+// when all listeners are HTTP-like.
```

Comment:
(later) Maybe dual-protocol ingress gateways should be rendered as N nodes in the UI instead of trying to mush them into one apples/oranges node like this.

Reply:
Yeah that's an option, can bring that up to Hannah and Jasmine